weather_app/src/Components/Weather.jsx

```diff
@@ -1,50 +1,63 @@
-import React, {useState} from 'react';
-import axios from 'axios'
+import React, { useState } from "react";
+import axios from "axios";
 
 const Weather = () => {
-    const [data, setData] = useState({})
-    const [location, setLocation] = useState('')
+  const [data, setData] = useState({});
+  const [location, setLocation] = useState("");
 
-    const searchLocation = (event) => {
-        if (event.key === 'Enter') {
-            axios.get(url).then((response) => {
-                setData(response.data)
-                console.log(response.data)
-            })
-            setLocation("")
-        }
+  const searchLocation = (event) => {
+    if (event.key === "Enter") {
+      // all async request should be handled with trycatch
```

Comment:
all async request should be handled with trycatch and handle error block
for example show another component that inform user that request failed
you have two types error
![Screenshot 2024-01-22 at 18 38 32](https://github.com/ViktorProjects/Weather_App/assets/106515539/d192e283-0fb0-42e5-8bd2-d777e0684547)
![Screenshot 2024-01-22 at 18 38 53](https://github.com/ViktorProjects/Weather_App/assets/106515539/9f885fdc-1aa9-4290-bde1-bec674ef2538)